### ISSUE TITLE
Not returning from "AcquireTokenInteractive" when SetAuthenticationContinuationEventArgs after cancel (backbutton)

### ISSUE THREAD

**Issue by laurensbro** [AzureAD/microsoft-authentication-library-for-dotnet#1661]
**Which Version of MSAL are you using ?** 4.9.0 - preview with broker support for android

**Platform** xamarin android

**What authentication flow has the issue?**
* Mobile
    * [v ] Interactive
   
**Is this a new or existing app?**
c. This is a new app or experiment

**Repro**

```csharp
// Calling 
var app = PublicClientApplicationBuilder.Create(msalClientId)
                .WithAuthority(EndpointConfiguration.Authority)
                .WithRedirectUri($"msauth://********")
                .WithBroker()
                .Build();

var authResult = await app
                        .AcquireTokenInteractive(GetADScopes())
                        .WithParentActivityOrWindow(parentActivity)
                        .WithUseEmbeddedWebView(true)
                        .ExecuteAsync();

var msalToken = authResult.AccessToken; //<-- Never getting here

//In mainactivity 
protected override void OnActivityResult(int requestCode, Result resultCode, Intent data)
{
            base.OnActivityResult(requestCode, resultCode, data);
AuthenticationContinuationHelper.SetAuthenticationContinuationEventArgs(requestCode,
                    resultCode, 
                    data);    
}
```

**Expected behavior**
Calling AcquireTokenInteractive, device starts MS-Authenticator as expected. After pressing "back" (android)button, the app goes to OnActivityResult with code 2001 (error / canceled?). I'm then calling SetAuthenticationContinuationEventArgs. Expected this to allow AcquireTokenInteractive to return an authResult.

**Actual behavior**
App never returns to the calling code.
Sidenote: No problem when signing in correctly, only after hitting back-button.

**Possible Solution**

**Additional context/ Logs / Screenshots**

**Comment by bgavrilMS** [comment]
This is possibly a bug.

**Comment by bgavrilMS** [comment]
The expected behaviour is to throw an exception with an `authentication_cancelled` error code.

**Comment by bgavrilMS** [comment]
I also logged bug https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/1663 to ensure that the behaviour on iOS and Android is consistent. 

Ideally it should also be consistent to when not using a broker, but this is a bit harder. When not using a broker, MSAL will throw an `MsalClientException` with error code `authentication_cancelled`

But we treat errors coming from the brokers as `MsalServiceException` 

@jennyf19 @trwalke - can you confirm that iOS and Android team also throw Service Exceptions when dealing with the broker? As @henrik-me mentioned a while back, this should really be a separate exception class `MsalBrokerException`

**Comment by trwalke** [comment]
released in 4.9.0
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.9.0

**Comment by iPaul-777** [comment]
Was this ever resolved?  I'm using Microsoft.Identity.Client 4.51.0 (latest version) in my Maui application with Android as my platform and this exact same thing is happening to me.  AcquireTokenInteractive is never returning.  So is this even possible?  Any work-arounds?
